### PR TITLE
Added keyboard navigation to Menu component.

### PR DESCRIPTION
**Pull request by TiagoJustino**
Added keyboard navigation (up and down arrow keys) to the Menu drop window.

### CODE REVIEW

File: src/js/components/Menu.js [hunk 1]
```diff
@@ -114,6 +114,34 @@ var Menu = React.createClass({
     this.setState({active: false});
   },
 
+  _onUpKeyPress: function (event) {
+    if (!this._drop.activeChild) {
+      var children = this._drop.container.querySelector('.box').childNodes;
```
Comment (alansouzati) [inline src/js/components/Menu.js]: what if there are more than one element with `.box` class? Can we assign an id to the box container? or something that is guaranteed to be unique?

File: src/js/components/Menu.js [hunk 2]
```diff
@@ -114,6 +114,34 @@ var Menu = React.createClass({
...
+      var lastChild = children[children.length - 1];
+      this._drop.activeChild = lastChild;
+    } else {
+      if (this._drop.activeChild.previousSibling) {
```
Comment (alansouzati) [inline src/js/components/Menu.js]: can we join these guys? `else if (...`>

File: src/scss/grommet-core/_objects.menu.scss [hunk 3]
```diff
@@ -91,6 +91,11 @@
         text-decoration: none;
         background-color: $hover-background-color;
       }
+
+      &:focus {
```
Comment (alansouzati) [inline src/scss/grommet-core/_objects.menu.scss]: can we join this with the hover ? 

```css
&:hover, 
&:focus {
  ...
```

File: src/js/components/Menu.js [hunk 4]
```diff
@@ -184,6 +214,16 @@ var Menu = React.createClass({
...
     if (! this.state.controlFocused && prevState.controlFocused) {
       this.stopListeningToKeyboard(focusedKeyboardHandlers);
+
+      // This was added because removing the focusedKeyboardHandlers also
+      // removes the `down` and `space` events from activeKeyboardHandlers.
+      // - Maybe this behaivor should be changed in the KeyboardAccelerators
+      // mixin itself?
+      // - Maybe the activeKeyboardHandlers and the focusedKeyboardHandlers
+      // should be added to two different components?
+      if (this.state.active) {
```
Comment (alansouzati) [inline src/js/components/Menu.js]: if you move this to be inside the mixin does it work?

File: src/js/components/Menu.js (hunk 1 above)
Comment (TiagoJustino) [inline src/js/components/Menu.js]: Done. Thanks for the heads up =)

File: src/js/components/Menu.js (hunk 2 above)
Comment (TiagoJustino) [inline src/js/components/Menu.js]: Done.

File: src/scss/grommet-core/_objects.menu.scss (hunk 3 above)
Comment (TiagoJustino) [inline src/scss/grommet-core/_objects.menu.scss]: I didn't know how I could join them. Thank you for teaching me that =)

File: src/js/components/Menu.js (hunk 4 above)
Comment (TiagoJustino) [inline src/js/components/Menu.js]: I changed the way how we manage the states a little bit. I thinks it's solved now (and the code is a little bit simpler too.)

File: src/js/components/Menu.js [hunk 5]
```diff
@@ -24,7 +25,8 @@ var MenuDrop = React.createClass({
     large: React.PropTypes.bool,
     onClick: React.PropTypes.func.isRequired,
     router: React.PropTypes.func,
-    small: React.PropTypes.bool
+    small: React.PropTypes.bool,
+    navContainerId: React.PropTypes.string.isRequired
```
Comment (alansouzati) [inline src/js/components/Menu.js]: why we need this here, if we generating a new one on line 243?

File: src/js/components/Menu.js [hunk 6]
```diff
@@ -107,19 +111,50 @@ var Menu = React.createClass({
 
   _onOpen: function (event) {
     event.preventDefault();
-    this.setState({active: true});
+    this.setState({state: 'expanded'});
```
Comment (alansouzati) [inline src/js/components/Menu.js]: i like having 3 states now. :+1:

File: src/js/components/Menu.js [hunk 7]
```diff
@@ -107,19 +111,50 @@ var Menu = React.createClass({
...
+
+  _onDownKeyPress: function (event) {
+    if (!this._drop.activeMenuItem) {
+      this._drop.activeMenuItem = document.getElementById(this._navContainerId).childNodes[0];
+    } else if (this._drop.activeMenuItem.nextSibling) {
+      this._drop.activeMenuItem = this._drop.activeMenuItem.nextSibling;
+    }
+
+    this._drop.activeMenuItem.focus();
+    return true;
```
Comment (alansouzati) [inline src/js/components/Menu.js]: can we add a comment that explain why `return true` is required here? I remember it is for stopPropagation, but it is good to explain it ;)

File: src/js/components/Menu.js [hunk 8]
```diff
@@ -148,8 +187,8 @@ var Menu = React.createClass({
       inline = (! this.props.label && ! this.props.icon);
     }
     return {
-      controlFocused: false,
-      active: false,
+      // state may be 'collapsed', 'focused' or 'expanded' (active).
```
Comment (alansouzati) [inline src/js/components/Menu.js]: I like that you care about explaining stuff :+1:

File: src/js/components/Menu.js [hunk 9]
```diff
@@ -271,9 +310,8 @@ var Menu = React.createClass({
     } else {
       onClick = this._onSink;
     }
-
     return (
-      <MenuDrop router={this.context.router}
+      <MenuDrop navContainerId={this._navContainerId} router={this.context.router}
```
Comment (alansouzati) [inline src/js/components/Menu.js]: Why we need this? It may be related to the first question on top...

File: src/js/components/Menu.js (hunk 5 above)
Comment (TiagoJustino) [inline src/js/components/Menu.js]: The same ID must be know by Menu and MenuDrop. The way I found was creating it on Menu and then passing it to MenuDrop. I accept suggestions =)

File: src/js/components/Menu.js (hunk 7 above)
Comment (TiagoJustino) [inline src/js/components/Menu.js]: Will do =)

File: src/js/components/Menu.js (hunk 5 above)
Comment (alansouzati) [inline src/js/components/Menu.js]: Maybe we can move the key event handling to be inside MenuDrop?

Comment (TiagoJustino) [inline src/js/components/Menu.js]: Awesome idea. It simplified the code =D

**Comment by alansouzati** [comment]
Awesome, thanks for your continued work on this!

**Pull request merged**